pkg/plugins/resources/hcl/main.go

```diff
@@ -0,0 +1,131 @@
+package hcl
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/minamijoyo/hcledit/editor"
+	"github.com/mitchellh/mapstructure"
+	"github.com/sirupsen/logrus"
+	"github.com/updatecli/updatecli/pkg/core/result"
+	"github.com/updatecli/updatecli/pkg/core/text"
+	"github.com/updatecli/updatecli/pkg/plugins/utils"
+)
+
+type Hcl struct {
+	spec             Spec
+	contentRetriever text.TextRetriever
+	files            map[string]file // map of file paths to file contents
+}
+
+type file struct {
+	originalFilePath string
+	filePath         string
+	content          string
+}
+
+func New(spec interface{}) (*Hcl, error) {
+	newSpec := Spec{}
+
+	err := mapstructure.Decode(spec, &newSpec)
+	if err != nil {
+		return nil, err
+	}
+
+	newResource := &Hcl{
+		spec:             newSpec,
+		contentRetriever: &text.Text{},
+	}
+
+	err = newResource.spec.Validate()
+	if err != nil {
+		return nil, err
+	}
+
+	newResource.files = make(map[string]file)
+	// File as unique element of newResource.files
+	if len(newResource.spec.File) > 0 {
+		filePath := strings.TrimPrefix(newResource.spec.File, "file://")
+		newResource.files[filePath] = file{
+			originalFilePath: filePath,
+			filePath:         filePath,
+		}
+	}
+	// Files
+	for _, filePath := range newResource.spec.Files {
+		filePath := strings.TrimPrefix(filePath, "file://")
+		newResource.files[filePath] = file{
+			originalFilePath: filePath,
+			filePath:         filePath,
+		}
+	}
+
+	return newResource, nil
+}
+
+func (h *Hcl) Query(resourceFile file) (string, error) {
+	query := h.spec.Key
+
+	sink := editor.NewAttributeGetSink(query)
+
+	inStream := strings.NewReader(resourceFile.content)
+	outStream := new(bytes.Buffer)
+	err := editor.DeriveStream(inStream, outStream, resourceFile.filePath, sink)
+	if err != nil {
+		return "", err
+	}
+
+	// When a value isn't found hcledit returns an empty byte array,
+	// where as result always has \n appended so will be > 0
+	if outStream.Len() == 0 {
+		err := fmt.Errorf("%s cannot find value for path %q from file %q",
+			result.FAILURE,
+			query,
+			resourceFile.originalFilePath)
+		return "", err
+	}
+
+	queryOutput := outStream.String()
+	queryOutput = strings.TrimSpace(queryOutput)
+	queryOutput = strings.Trim(queryOutput, "\"")
+	return queryOutput, nil
+}
+
+// Read puts the content of the file(s) as value of the y.files map if the file(s) exist(s) or log the non existence of the file
+func (h *Hcl) Read() error {
+	var err error
+
+	// Retrieve files content
+	for filePath := range h.files {
+		f := h.files[filePath]
+		if h.contentRetriever.FileExists(f.filePath) {
+			f.content, err = h.contentRetriever.ReadAll(f.filePath)
```

Comment:
I really love the fact you reused the `contentReatriever` like the other file resource.
This means we can also using other supported protocols like https.
I just tested the following snippets and it works.
It could be worth to add to the e2e tests

```
conditions
  https:
    name: Condition using value
    kind: hcl 
    sourceid: scenario1
    spec:
      file: https://raw.githubusercontent.com/mcwarman/updatecli/hcl/pkg/plugins/resources/hcl/testdata/data.hcl
      key: resource.person.john.surname
      value: Doe 
```

Reply:
using a file from https://github.com/updatecli-test/jenkins-infra-aws

Reply:
Yeah i noticed it when i was looking into it more last night, so re-wrote the initial version to make the most of the feature.